pkgs/top-level/ocaml-packages.nix

```diff
@@ -17,6 +17,8 @@ let
 
     buildOcaml = callPackage ../build-support/ocaml { };
 
+    dunePackage = callPackage ../build-support/ocaml/dune.nix {};
```

Comment:
I would prefer `buildDunePackage` to be consistent with all other language abstraction such as `buildGoPackage`, `buildPythonLibrary` etc. Also since you are introducing a new abstraction there should be documentation with an example in `nixpkgs/doc/languages-frameworks`

Reply:
“build” in nix parlance refers to something else. This function does not build anything. It creates or makes a derivation, but does not build it.

You are right that there are other functions with misleading names in nixpkgs, though.

Reply:
Sorry this is way to too late to change. We are using this scheme EVERYWHERE.
Not using it here would create even more confusion.

Reply:
`mkDerivation` does not read “build”.

Reply:
- buildFHSUserEnv
- buildGoPackage
- buildRubyGem
- buildEnv
- buildRPackage
- buildEmscriptenPackage
- buildPerlPackage
- buildMaven
- buildHaskellPackages
- buildPythonApplication
- buildPythonLibrary
- buildDotnetPackage
- dockerTools.buildImage

These function all have in common that they change what the `builder` does hence the name.

Reply:
Thanks for the explanation. I still don’t like that naming scheme, but that’s only my opinion.

Reply:
So @vbgl, should I rename to `buildDunePackage`?

Reply:
Otherwise, what about `mkDuneDerivation`? It conveys the meaning better, I suppose, even, if it is less close to the convention (and conventions can change, it is never "too late"). After all, `mkDerivation` is itself based on the built-in `derivation`. If it was respecting the convention, it would be called `buildGeneric` or something like this.

Reply:
I think `mkDerivation` is not good name at all for people that have not 
read the nix paper or studied nix pills.
From the point of a nixpkgs newcomer `buildDunePackage` has a meaning while `mkDuneDerivation` sounds obscure.

Reply:
Well, right, although I can't tell if you can go very far using Nix without having read nix pills. Anyways, for the sake of advancing this PR, I will now go ahead and apply the name change you suggested.